android/libraries/rib-base/src/main/java/com/badoo/ribs/android/recyclerview/Adapter.kt

```diff
@@ -60,42 +65,49 @@ internal class Adapter<T : Parcelable>(
 
     override fun onBindViewHolder(holder: ViewHolder, position: Int) {
         val entry = items[position]
-        holder.configurationKey = entry.configurationKey
-        holder.uuid = entry.uuid
-
+        holder.identifier = entry.identifier
     }
 
     override fun onViewAttachedToWindow(holder: ViewHolder) {
         super.onViewAttachedToWindow(holder)
-        val configurationKey = holder.configurationKey!! // at this point it should be bound
+        val identifier = holder.identifier!! // at this point it should be bound
+        holders[identifier] = WeakReference(holder)
+
         if (hostingStrategy == LAZY) {
-            router.add(configurationKey)
-        }
-        router.activate(configurationKey)
-        router.getNodes(configurationKey)!!.forEach { childNode ->
-            childNode.attachToView(holder.itemView as FrameLayout)
+            val entry = feature.state.items.find { it.identifier == identifier }!!
+            routingSource.add(entry.element, entry.identifier)
         }
+
+        routingSource.activate(identifier)
+    }
+
+    override fun activate(routing: Routing<T>, child: Node<*>) {
+        holders[routing.identifier]?.get()?.let { holder ->
+            child.attachToView(holder.itemView as FrameLayout)
+        } ?: RIBs.errorHandler.handleNonFatalError("Holder is gone! Routing: $routing, child: $child")
     }
 
     override fun onViewRecycled(holder: ViewHolder) {
         super.onViewRecycled(holder)
-        val configurationKey = holder.configurationKey!! // at this point it should be bound
-        deactivate(configurationKey)
+        val identifier = holder.identifier!! // at this point it should be bound
```

Comment:
may be better to be more safer here? and just add let with errorHandler if there is no such?